src/main/java/com/adamk33n3r/runelite/watchdog/EventHandler.java

```diff
@@ -189,26 +191,54 @@ private void onItemContainerChanged(ItemContainerChanged itemContainerChanged) {
-                    this.fireAlert(inventoryAlert, inventoryAlert.getInventoryAlertType().getName());
-                } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.ITEM) {
-                    Map<Integer, Integer> allItems = new HashMap<>();
-                    Arrays.stream(items)
-                        .forEach(item -> allItems.merge(item.getId(), item.getQuantity(), Integer::sum));
-                    allItems.entrySet().stream()
-                        .filter(itemWithCount -> inventoryAlert.getItemQuantity() == 0 || itemWithCount.getValue() == inventoryAlert.getItemQuantity())
-                        .map(itemWithCount -> this.matchPattern(inventoryAlert, this.itemManager.getItemComposition(itemWithCount.getKey()).getName()))
-                        .filter(Objects::nonNull)
-                        .findFirst()
-                        .ifPresent(groups -> this.fireAlert(inventoryAlert, groups));
-                }
-            });
+        Item[] items = itemContainerChanged.getItemContainer().getItems();
+        long itemCount = Arrays.stream(items).filter(item -> item.getId() > -1).count();
+        Map<Integer, Integer> allItems = new HashMap<>();
+        Arrays.stream(items)
+                .forEach(item -> allItems.merge(item.getId(), item.getQuantity(), Integer::sum));
+        // Skip firing alerts if there are no previous items, since we just logged in. Even an empty inventory will have a map of -1 itemIds.
+        if (!this.previousItemsTable.isEmpty()) {
+            this.alertManager.getAllEnabledAlertsOfType(InventoryAlert.class)
+                    .forEach(inventoryAlert -> {
+                        if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.FULL && itemCount == 28) {
+                            this.fireAlert(inventoryAlert, inventoryAlert.getInventoryAlertType().getName());
+                        } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.EMPTY && itemCount == 0) {
+                            this.fireAlert(inventoryAlert, inventoryAlert.getInventoryAlertType().getName());
+                        } else if (inventoryAlert.getInventoryAlertType() == InventoryAlert.InventoryAlertType.ITEM) {
+                            allItems.entrySet().stream()
+                                    .filter(itemWithCount -> itemWithCount.getKey() != -1 && (inventoryAlert.getItemQuantity() == 0 || itemWithCount.getValue() == inventoryAlert.getItemQuantity()))
+                                    .map(itemWithCount -> this.matchPattern(inventoryAlert,
+                                            this.itemManager.getItemComposition(itemWithCount.getKey()).getName()))
+                                    .filter(Objects::nonNull)
+                                    .forEach(groups -> this.fireAlert(inventoryAlert, groups));
```

Comment:
> I guess this handles the case where a regex matches multiple items and we'd want to fire for each?

Yup! Exactly. This wouldn't fire twice for the same item since they are already deduped by the map. So this would only affect a name pattern which matches multiple.